### PR TITLE
Create getHistory endpoint.

### LINKED ISSUE #13: Create an assetsList endpoint to return the user's uploads
This is need it on the frontend to display all the assets uploaded by an user. The information we need can follow what's in being returned in /getQuote + status + asset's info

{ "tokenAmount": 500, "approveAddress": "0x123", "chainId": 1, "tokenAddress": "0xOCEAN_on_MAINNET", "quoteId": "xxxx", "status": 400, "cid": "xxx" }

Related issue: oceanprotocol/decentralized_storage_backend#45

### PR DESCRIPTION
**Pull request by mariacarmina**
Fixes #13  .

Changes proposed in this PR:

- create getHistory endpoint

### CODE REVIEW

**Comment by mariacarmina** [comment]
Needs testing, still WIP

**Comment by mariacarmina** [comment]
Fixed endpoint & tests.
Example of history output:

```console

{
    quoteId: 'f4b4de9d5edbffd79f76e626b30919dd',
    status: 403,
    chainId: 80001,
    tokenAddress: '0x9c3C9283D3e44854697Cd22D3Faa240Cfb032889',
    tokenAmount: '936092174560178',
    approveAddress: '0xE5e4b9781BBe23fD95595F1D9fA6FD5C986F5378',
    transactionHash: 'KvL_lPDEbRMUEgEtor97IBWhBjI5ycGVQk2pt0oB700'
  },
  {
    quoteId: '1f4a53aa86103ced3185867445588457',
    status: 400,
    chainId: 80001,
    tokenAddress: '0x9c3C9283D3e44854697Cd22D3Faa240Cfb032889',
    tokenAmount: '936133766348194',
    approveAddress: '0xE5e4b9781BBe23fD95595F1D9fA6FD5C986F5378',
    transactionHash: 'sdzlfXsAWxQdfVSXLBJWHOQcRtvHe7S_ca-YLz55BqE'
  },
  {
    quoteId: '1f4a53aa86103ced3185867445588457',
    status: 400,
    chainId: 80001,
    tokenAddress: '0x9c3C9283D3e44854697Cd22D3Faa240Cfb032889',
    tokenAmount: '936133766348194',
    approveAddress: '0xE5e4b9781BBe23fD95595F1D9fA6FD5C986F5378',
    transactionHash: 'z8T_fmbLzNLPOSdTN-s02W-bOnKE4XeriV9-70szy74'
  }

```

File: app/controllers/quote.controller.js [hunk 1]
```diff
@@ -414,3 +414,80 @@ exports.getLink = async (req, res) => {
...
+	let signerAddress;
+	try {
+		signerAddress = ethers.utils.verifyMessage(message, signature);
+	}
+	catch(err) {
+		errorResponse(req, res, err, 403, "Invalid signature.");
+		return;
+	}
+
+	if(signerAddress != userAddress) {
```
Comment (jamiehewitt15) [inline app/controllers/quote.controller.js:456]:
```suggestion
	if(signerAddress !== userAddress) {
```

File: app/controllers/quote.controller.js [hunk 2]
```diff
@@ -414,3 +414,80 @@ exports.getLink = async (req, res) => {
...
+		Nonce.set(userAddress, nonce);
+	}
+	catch(err) {
+		errorResponse(req, res, err, 500, "Error occurred while setting nonce.");
+		return;
+	}
+
+	try {
+		const history = Quote.getHistory(userAddress);
+		if(history == undefined) {
```
Comment (jamiehewitt15) [inline app/controllers/quote.controller.js:483]:
```suggestion
		if(history === undefined) {
```

**Review by jamiehewitt15 — APPROVED** [review]
Looks good, just a few small suggestions

**Pull request merged**